src/test/utils/overridesHelper.ts

```diff
@@ -1,61 +1,101 @@
+import * as overrides from "#app/overrides";
 
 /**
  * Helper to handle overrides in tests
  */
 export class OverridesHelper {
-  constructor() {}
+  game: GameManager;
+  constructor(game: GameManager) {
+    this.game = game;
+  }
 
   /**
    * Override the encounter chance for a mystery encounter.
    * @param percentage the encounter chance in %
+   * @returns spy instance
    */
-  mysteryEncounterChance(percentage: number) {
+  mysteryEncounterChance(percentage: number): MockInstance {
     const maxRate: number = 256; // 100%
     const rate = maxRate * (percentage / 100);
-    vi.spyOn(Overrides, "MYSTERY_ENCOUNTER_RATE_OVERRIDE", "get").mockReturnValue(rate);
+    const spy = vi.spyOn(Overrides, "MYSTERY_ENCOUNTER_RATE_OVERRIDE", "get").mockReturnValue(rate);
     this.log(`Mystery encounter chance set to ${percentage}% (=${rate})!`);
+    return spy;
+  }
+
+  /**
+   * Override the encounter that spawns for the scene
+   * @param encounterType
+   * @returns spy instance
+   */
+  mysteryEncounter(encounterType: MysteryEncounterType): MockInstance {
+    const spy = vi.spyOn(overrides, "MYSTERY_ENCOUNTER_OVERRIDE", "get").mockReturnValue(encounterType);
+    this.log(`Mystery encounter override set to ${encounterType}!`);
+    return spy;
   }
 
   /**
    * Override the starting biome
-   * @warning The biome will not be overridden unless you call `workaround_reInitSceneWithOverrides()` (testUtils)
+   * @warning Any event listeners that are attached to [NewArenaEvent](events\battle-scene.ts) may need to be handled down the line
    * @param biome the biome to set
    */
   startingBiome(biome: Biome) {
-    vi.spyOn(Overrides, "STARTING_BIOME_OVERRIDE", "get").mockReturnValue(biome);
+    this.game.scene.newArena(biome);
     this.log(`Starting biome set to ${Biome[biome]} (=${biome})!`);
   }
 
   /**
    * Override the starting wave (index)
    * @param wave the wave (index) to set. Classic: `1`-`200`
+   * @returns spy instance
    */
-  startingWave(wave: number) {
-    vi.spyOn(Overrides, "STARTING_WAVE_OVERRIDE", "get").mockReturnValue(wave);
+  startingWave(wave: number): MockInstance {
+    const spy = vi.spyOn(Overrides, "STARTING_WAVE_OVERRIDE", "get").mockReturnValue(wave);
     this.log(`Starting wave set to ${wave}!`);
+    return spy;
+  }
+
+  /**
+   * Override each wave to have or not have standard trainer battles
+   * @returns spy instance
+   * @param isTrainer
+   */
+  trainerWave(isTrainer: boolean): MockInstance {
+    const spy = vi.spyOn(this.game.scene.gameMode, "isWaveTrainer").mockReturnValue(isTrainer);
+    this.log(`${isTrainer? "forcing" : "ignoring"} trainer waves!`);
+    return spy;
   }
 
   /**
    * Override the weather (type)
    * @param type weather type to set
+   * @returns spy instance
    */
-  weather(type: WeatherType) {
-    vi.spyOn(Overrides, "WEATHER_OVERRIDE", "get").mockReturnValue(type);
+  weather(type: WeatherType): MockInstance {
```

Comment:
```suggestion
  weather(type: WeatherType) {
```


let it self-inject the type. That is a more specific version of MockInstance